### PR TITLE
MGMT-16976 Resource syncs pages

### PR DESCRIPTION
**Pull request by celdrake**
https://issues.redhat.com/browse/MGMT-16976

Manages Repository resource syncs.

rs-page

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -38,6 +38,17 @@ if (fs.existsSync('certs/api-sig.key')) {
 const cert = fs.readFileSync('certs/front-cli.crt');
 const certkey = fs.readFileSync('certs/front-cli.key');
 const ca = fs.readFileSync('certs/ca.crt');
+
+const buildUrl = (resource, query) => {
```
Comment (rawagner) [inline app.js:42]: lets do this in frontend code

File: src/app/components/Repository/RepositoryDetails.tsx [hunk 2]
```diff
@@ -24,9 +24,10 @@ import { Repository } from '@types';
 
 import StatusInfo from '@app/components/common/StatusInfo';
 import { getRepositoryLastTransitionTime, getRepositorySyncStatus } from '@app/utils/status/repository';
+import RepositoryResourceSyncList from '@app/components/ResourceSync/RepositoryResourceSyncList';
 
 const RepositoryDetails = () => {
-  const { repositoryId } = useParams();
+  const { repositoryId } = useParams() as { repositoryId: string };
```
Comment (rawagner) [inline src/app/components/Repository/RepositoryDetails.tsx:30]:
```suggestion
  const { repositoryId } = useParams<{ repositoryId: string }>();
```

File: src/app/components/ResourceSync/RepositoryResourceSyncList.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+          </Tr>
+        ))}
+      </Tbody>
+    </Table>
+  );
+};
+
+const RepositoryResourceSyncList = ({ repositoryId }: { repositoryId: string }) => {
+  const [isFormVisible, setIsFormVisible] = React.useState<boolean>(false);
+  const [rsList, , , refetch] = useFetchPeriodically<ResourceSyncList>({
```
Comment (rawagner) [inline src/app/components/ResourceSync/RepositoryResourceSyncList.tsx:60]: Im missing loading/error state handling

**Review by rawagner — CHANGES_REQUESTED** [review]

File: src/app/components/Repository/RepositoryDetails.tsx (hunk 2 above)
Comment (celdrake) [reply src/app/components/Repository/RepositoryDetails.tsx:30]: The definition of `useParams` allows for `string | undefined` , so this suggestion is not equivalent.
With it, on line 106 we would have to cast `repositoryId`, or provide a fallback for it.

File: src/app/components/ResourceSync/RepositoryResourceSyncList.tsx (hunk 3 above)
Comment (celdrake) [reply src/app/components/ResourceSync/RepositoryResourceSyncList.tsx:60]: I'm adding a basic handling of both.

File: src/app/components/Repository/RepositoryDetails.tsx (hunk 2 above)
Comment (rawagner) [reply src/app/components/Repository/RepositoryDetails.tsx:30]: oh, that's a shame :/

File: app.js (hunk 1 above)
Comment (celdrake) [reply app.js]: Checked after rebasing, and the query using `labelSelector` works fine with the new Go proxy, no need for these adaptations anymore.

**Review by rawagner — APPROVED** [review]

**Pull request merged**